monkey/monkey_island/cc/ui/src/components/ui-components/AdvancedMultiSelect.js

```diff
@@ -1,33 +1,24 @@
-import React, {useState} from 'react';
-
+import React from "react";
 import {Card, Button, Form} from 'react-bootstrap';
 import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
 import {faCheckSquare} from '@fortawesome/free-solid-svg-icons';
+import {faMinusSquare} from '@fortawesome/free-solid-svg-icons';
 import {faSquare} from '@fortawesome/free-regular-svg-icons';
 import {cloneDeep} from 'lodash';
 
 import {getComponentHeight} from './utils/HeightCalculator';
 import {resolveObjectPath} from './utils/ObjectPathResolver';
 import InfoPane from './InfoPane';
 
-
-function getSelectValuesAfterClick(valueArray, clickedValue) {
-  if (valueArray.includes(clickedValue)) {
-    return valueArray.filter((e) => {
-      return e !== clickedValue;
-    });
-  } else {
-    valueArray.push(clickedValue);
-    return valueArray;
-  }
+const MasterCheckboxState = {
+  NONE: 0,
+  MIXED: 1,
+  ALL: 2
```

Comment:
Shouldn't be named ALL, should be DEFAULT

Reply:
See explanation of behavior changes in the PR description.